EE.C

```diff
@@ -0,0 +1,478 @@
+	  
+	  //pT
+	  float pt1 = Lep_pt[0];
+	  float pt2 = Lep_pt[1];
+
+
+	  leppT_DY->Fill(pt1);
+	  leppT_DY->Fill(pt2);
+
+	  //Iso
+		  eleIso_DY->Fill(Lep_Iso[1]);
+		  eleIso_DY->Fill(Lep_Iso[0]);
+	 
+
+	  //e-mu dPhi and dR, e-mu mass
+	  dPhi_DY->Fill(Dil_dPhi[0]);
+	  dR_DY->Fill(Dil_dR[0]);
+	  ee_mass_DY->Fill(Dil_mass[0]);
+
+	  //eta
+	  float eta1 = Lep_eta[0];
+	  float eta2 = Lep_eta[1];
+
+
+	  lepeta_DY->Fill(eta1);
+	  lepeta_DY->Fill(eta2);
+
+	  //phi
+	  float phi1 = Lep_phi[0];
+	  float phi2 = Lep_phi[1];
+
+
+	  lepphi_DY->Fill(phi1);
+	  lepphi_DY->Fill(phi2);
+
+	 //leading and 2nd leading declaration 
+	  float leadjet_pT = -999;
+	  float secjet_pT = -999;
+
+	  //dR(electron,b-jet)..what is closest electron?
+	  //float EleBjet_dR;
+
+//		    btagEvent_DY->Fill(lep_pt[0]);
+//		    btagEvent_DY->Fill(lep_pt[1]);
+
+		    //jet pT and jet eta(if Number of b-jet = 1, it is possible to use)
+		    for(int k=0; k<Jet_pt.size(); k++){
+			
+		    jetpT_DY->Fill(Jet_pt[k]);
+		    jeteta_DY->Fill(Jet_eta[k]);
+		    }
+
+
+		    //all jet pT sum^2
+		    float sum_jetpT = 0.;
+
+		    //leading jet
+		    for(int j=0; j<bJet_pt.size(); j++){
+		        bjetpT_DY->Fill(bJet_pt[j]); // all b jet pT
+		        bjeteta_DY->Fill(bJet_eta[j]); // all b jet pT
+
+			//all jet pT sum
+			sum_jetpT = sum_jetpT + sqrt(Jet_pt[j]*jet_pt[j]);
+
+
+			//b-jet pT vs dR
+			float dphi = Lep_phi[1] - bJet_phi[j];
+			float deta = Lep_eta[1] - bJet_eta[j];
+			if(dphi < -TMath::Pi()) dphi = dphi + 2.*TMath::Pi();
+			if(dphi >= TMath::Pi()) dphi = dphi - 2.*TMath::Pi();
+			float EleBjet_dR = sqrt(pow(dphi,2) + pow(deta,2));//Delta R(electron,b-jet)...what is closect electron?
+			//bjetpT_dR_DY->Fill(bJet_pt[j],EleBjet_dR);
+
+			//b-jet pT vs electron pT
+			//bjetpT_elepT_DY->Fill(bJet_pt[j],Lep_pt[1]);
+
+			//b-jet pT vs muon pT
+			//bjetpT_mupT_DY->Fill(bJet_pt[j],Lep_pt[0]);
+			
+			if(bJet_pt[j] >= leadjet_pT) leadjet_pT = bJet_pt[j];
+			else leadjet_pT = leadjet_pT;
+		    }
+	      
+		    if(leadjet_pT == -999) continue;
+		    //2nd leading jet
+	  	    for(int u=0; u<bJet_pt.size(); u++){
+		  //if(jet_pt[i] >= leadjet_pT) continue;
+		  	if(secjet_pT < bJet_pt[u] && bJet_pt[u] < leadjet_pT) secjet_pT = bJet_pt[u];
+			else secjet_pT = secjet_pT;
+		    }
+	      
+	  //if(leadjet_pT < 30) continue;
+	  //if(secjet_pT < 30) continue;
+	      
+
+      //SBM
+
+            if(Nbjet == 1 || Nbjet == 2){
+
+                int Jet2_nonB=-1;
+                float Jet2_nonB_PT=0.;
+
+                for(unsigned Nj=0; Nj<Jet_pt.size(); ++Nj){
+                    if(Jet2_nonB_PT<Jet_pt[Nj]){
+                        if(Jet2_nonB_PT != bJet_pt[0]){
+                            Jet2_nonB=Nj;
+                            Jet2_nonB_PT=jet_pt[Nj];
+                        }
+                    }
+                }
+
+                //const double muon_mass = 0.1056583715;
+                const double ele_mass = 0.0005109989461;
```

Comment:
If you need that mass, you can make it dependent upon the lepton ID to generalize this without having to switch codes between dielectron and dimuon.